### PR TITLE
[GEOS-11141] Add section about logging configuration hardening

### PR DESCRIPTION
**Pull request by anthares101**
[GEOS-11141](https://osgeo-org.atlassian.net/browse/GEOS-11141) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

The idea is to add a little section to the documentation that can help users ensure the logging configuration is not missconfigured with neferious intentions even with administration access.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [ ] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Review by mprins — CHANGES_REQUESTED** [review]
Several typos and syntax error. This is not a markdown file.
Also I think it's poor and incomplete advice

**Comment by anthares101** [comment]
> Several typos and syntax error. This is not a markdown file. Also I think it's poor and incomplete advice

Sorry about the typos not english native speaker, I tried to fix both the typos and the syntax problem (I hope, did not use this format before). 

About the poor advice, the idea until the security advisory is published is to keep this vague. I can add more about the impact of the log configuration but I dont want to disclouse anything non public.

**Comment by aaime** [comment]
Idea for rewording:

```
For production systems, it is advised to set the log location via the environment variable. This prevents overriding it via the administration GUI or REST API, providing better separation of responsibility between those setting up and controlling the actual machine, and those configuring the GeoServer data, styles and the like.  Follow :ref:`overriding_the_log_location_setup_in_the_geoserver_configuration` for more information
```

**Comment by anthares101** [comment]
Thanks for the help @aaime 😄

File: doc/en/user/source/production/config.rst [hunk 1]
```diff
@@ -10,6 +10,11 @@ Logging may visibly affect the performance of your server. High logging levels a
 
 You can change the logging level in the :ref:`config_globalsettings_log_profile`.  You will want to choose the ``PRODUCTION`` logging configuration, where only problems are written to the log files.
 
+Logging configuration hardening
+-------------------------------
+
+For production systems, it is advised to set the log location via the environment variable. This prevents overriding it via the administration GUI or REST API, providing better separation of responsibility between those setting up and controlling the actual machine, and those configuring the GeoServer data, styles and the like. Follow :ref:`overriding_the_log_location_setup_in_the_geoserver_configuration` for more information
+
```
Comment (jodygarnett) [inline doc/en/user/source/production/config.rst:17]: Thanks, that is clearly written.

There are actually several ways (environmental variable, java system property, or web application parameter) to set these things. I know most system admins would just use the environmental variable, but there may still be someone configuring web application parameters in tomcat (hint I am that someone).

aside: I believe there are a number of settings that can be defined in this manner that all could be used if one wished to enforce a strong separation of concerns between those running the machine and the REST / GUI interface.

File: doc/en/user/source/production/config.rst [hunk 2]
```diff
@@ -10,6 +10,11 @@ Logging may visibly affect the performance of your server. High logging levels a
 
 You can change the logging level in the :ref:`config_globalsettings_log_profile`.  You will want to choose the ``PRODUCTION`` logging configuration, where only problems are written to the log files.
 
+Logging configuration hardening
+-------------------------------
+
+For production systems, it is advised to set the log location via the environment variable. This prevents overriding it via the administration GUI or REST API, providing better separation of responsibility between those setting up and controlling the actual machine, and those configuring the GeoServer data, styles and the like. Follow :ref:`overriding_the_log_location_setup_in_the_geoserver_configuration` for more information
```
Comment (jodygarnett) [inline doc/en/user/source/production/config.rst:16]:
```suggestion
For production systems, it is advised to define the log location using the ``GEOSERVER_LOG_LOCATION `` parameter (defined as either an environment variable, java system property, or servlet context parameter.)

Defining ``GEOSERVER_LOG_LOCATION`` has priority, causing the value provided using the Admin Console GUI or REST API to be ignored. This apporach establishes a separation of responsibility between those setting up and controlling the actual machine, and those configuring the GeoServer application.

Follow :ref:`overriding_the_log_location_setup_in_the_geoserver_configuration` for more information
```

**Review by jodygarnett — CHANGES_REQUESTED** [review]
I provided a clarification and broke your statement into smaller sentences.

For some reason the phrase "This prevents overriding" sounded awkward to me. It is more that the GEOSERVER_LOG_LOCATION has priority, or establishes the default value. 

You can consider the proposed changes and let me know what you think :)

I also looked if there was any other settings that could be handled in this manner, ``PROXY_BASE_URL`` was the only one that works in a similar fashion.

**Comment by anthares101** [comment]
@jodygarnett looking good, proposed changes applied 😄

**Comment by jodygarnett** [comment]
Got build failure:

05:11:06,747 [INFO]      [exec] /home/runner/work/geoserver/geoserver/doc/en/user/source/production/config.rst:16: WARNING: Inline literal start-string without end-string.
05:11:25,374 [INFO]      [exec] /home/runner/work/geoserver/geoserver/doc/en/user/source/production/config.rst:20: WARNING: undefined label: 'overriding_the_log_location_setup_in_the_geoserver_configuration'

**Review by mprins — CHANGES_REQUESTED** [review]
there is a build failure on this change:

```
05:11:06,747 [INFO]      [exec] /home/runner/work/geoserver/geoserver/doc/en/user/source/production/config.rst:16: WARNING: Inline literal start-string without end-string.
05:11:25,374 [INFO]      [exec] /home/runner/work/geoserver/geoserver/doc/en/user/source/production/config.rst:20: WARNING: undefined label: 'overriding_the_log_location_setup_in_the_geoserver_configuration'
```

File: doc/en/user/source/production/config.rst [hunk 3]
```diff
@@ -10,6 +10,15 @@ Logging may visibly affect the performance of your server. High logging levels a
 
 You can change the logging level in the :ref:`config_globalsettings_log_profile`.  You will want to choose the ``PRODUCTION`` logging configuration, where only problems are written to the log files.
 
+Logging configuration hardening
+-------------------------------
+
+For production systems, it is advised to set ``GEOSERVER_LOG_LOCATION `` parameter during startup. The value may be defined as either an environment variable, java system property, or servlet context parameter.
+
+Defining ``GEOSERVER_LOG_LOCATION`` has priority, causing the value provided using the Admin Console GUI or REST API to be ignored. This apporach establishes a separation of responsibility between those setting up and controlling the actual machine, and those configuring the GeoServer application.
```
Comment (bradh) [inline doc/en/user/source/production/config.rst:18]:
```suggestion
Defining ``GEOSERVER_LOG_LOCATION`` has priority, causing the value provided using the Admin Console GUI or REST API to be ignored. This approach establishes a separation of responsibility between those setting up and controlling the actual machine, and those configuring the GeoServer application.
```

File: doc/en/user/source/production/config.rst [hunk 4]
```diff
@@ -10,6 +10,15 @@ Logging may visibly affect the performance of your server. High logging levels a
...
+
+Follow :ref:`overriding_the_log_location_setup_in_the_geoserver_configuration` for more information.
```
Comment (jodygarnett) [inline doc/en/user/source/production/config.rst]:
```suggestion
Follow :ref:`Overriding the log location <logging>` for more information.
```

**Comment by jodygarnett** [comment]
Build failure fixed, we are good to go. Thanks for working on this everyone.

**Pull request merged**